docs/README.md

```diff
@@ -36,4 +36,9 @@ to libvirt domains based on the state of those resources.
  * `make cluster-sync`: After deploying a fresh environment, or after making
    changes to code in this tree, this command will sync the Pods and DaemonSets
    in the running KubeVirt environment with the state of this tree.
+   Note: Use make cluster-sync PARALLEL_CLEANUP=1, for a faster deploy.
+   Usable for healthy clusters and for first deploy only.
```

Comment:
What does this mean? What is the difference to a normal cleanup?

Reply:
normal clean run on each element and delete it forcefully , whatever it exists healty or not / not exists 
This script checks a preliminary conditon if there is a namespace of kubevirt (according the right env var).
if so it runs the whole script parallel to the build, and wait for both to finish before deploying.
if it doesnt exists it wont run the clean at all
so lets say you have a broken cluster, with a pod which is stuck on terminating, and the ns of kubevirt was already deleted,
then you will need to run manually clean, or use regular `make cluster-sync` which is untouched and does it anyway.

it cant happen on prow because each job runs on a fresh cluster-up, which doesnt have kubevirt deployed.
it can just happen due to a bug when a developer destroyed his already deployed cluster in the manner above (kubevirt ns doesnt exists (whatever the name of it), but zombie leftovers exists).

Reply:
any update please?

Reply:
Can't you just split the cleanup in two steps? The cleanup script is pretty much non-blocking. You can just delete everything (including the namespaces), then do the build and before you deploy just wait for the namespaces to really disappear.

Reply:
i wont be able at the moment to work on it, if possible please to do what you suggest in a follow PR,
because the method i created doesn't present a regression (non healthy cluster is a corner case which doesn't happen usually), and it does save time?

Reply:
what about the following?
support 3 options
1. dont run clean at all - can be used for first deployment where there is no cluster before.
2. run clean parallel
3. run clean regular as its now

Reply:
As explained before, the api delete calls are not blocking. where you are mostly waiting is that the namespaces actually disappear. No special cases means the least amount of chance to fail on not-clean clusters (for whatever reason) during development and makes the code most robust. Just delete first, then build and then wait for the namespaces to disappear.

Reply:
please see https://github.com/kubevirt/kubevirt/pull/6230